src/components/GenerateModal/GenerateModal.tsx

```diff
@@ -0,0 +1,296 @@
+import ReactDOM from 'react-dom';
+import SearchBar from 'pages/GroupChatList/HeaderLayout/SearchBar';
+import {MdClose} from 'react-icons/md';
+import styled from 'styled-components';
+import {theme} from 'styles/Theme';
+import UserCells from './UserCells';
+import {User} from 'types/chatroom.types';
+import {useEffect, useState} from 'react';
+import {getUsers} from 'api/users';
+import {useRecoilValue} from 'recoil';
+import {userInformation} from 'states/atom';
+import {useParams} from 'react-router';
+import {inviteChatRoom} from 'api/myChatRoom';
+import {useChatCreation} from 'hooks/useChatRoomMutation';
+
+interface ModalProps {
+  onToggleModal: React.Dispatch<boolean>;
+  headline: string;
+  label1: string;
+  optionLabel?: string;
+  optionInput: boolean;
+  label2: string;
+  primaryBtn: string;
+}
+
+type InputStates = [{name: 'chatName'; state: string}, {name: 'pickedUser'; state: string}];
+
+const GenerateModal = (props: ModalProps) => {
+  const [userData, setUserData] = useState<[User[], User[]]>([[], []]);
+  // [선택가능한 유저 배열, 선택된유저 배열]
+  const [searchUserData, setSearchUserData] = useState('');
+  // 검색어
+  const [chatName, setChatName] = useState<string>('');
+  // 방이름
+  const [inputStates, setInputStates] = useState<InputStates>([
+    {name: 'chatName', state: 'default'},
+    {name: 'pickedUser', state: 'default'},
+  ]);
+  // 에러 상태
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  // 로딩 중 (비동기 요청 중)
+
+  const {chatId} = useParams();
+  const myInfo = useRecoilValue(userInformation);
+  const {createChatRoom} = useChatCreation();
+
+  useEffect(() => {
+    setIsLoading(true);
+    // 사용자 불러오기
+    getUsers()
+      .then(res => {
+        const filtered = (res as User[]).slice().filter(val => val.id !== myInfo.id);
+        setUserData([filtered, []]);
+      })
+      .catch(err => console.log(err))
+      .finally(() => setIsLoading(false));
+  }, [myInfo.id]);
```

Comment:
마운트때만 사용자를 불러오는 것 같은데,
이럴 땐 요 useEffect의 역할을 함수명으로 명시하는 것도 협업에 좋습니다!

```suggestion
  useEffect(function getUserDataOnMount() {
    setIsLoading(true);
    // 사용자 불러오기
    getUsers()
      .then(res => {
        const filtered = (res as User[]).slice().filter(val => val.id !== myInfo.id);
        setUserData([filtered, []]);
      })
      .catch(err => console.log(err))
      .finally(() => setIsLoading(false));
  }, [myInfo.id]);
```